src/GitHub.Api/Git/GitClient.cs

```diff
@@ -148,9 +152,39 @@ private Task<NPath> LookForPortableGit()
             return path;
         }
 
-        public bool ValidateGitInstall(NPath path)
+        public ITask<ValidateGitInstallResult> ValidateGitInstall(string path)
         {
-            return path.FileExists();
+            if (!path.ToNPath().FileExists())
+            {
+                return new FuncTask<ValidateGitInstallResult>(cancellationToken, 
+                    () => new ValidateGitInstallResult(false, null, null));
```

Comment:
This should probably be one of those Completed tasks that we never know the name of?

Reply:
`TaskHelpers.GetCompletedTask` returns a `Task<T>` but I need `ITask<T>`. We dont have a `GetCompletedTask` method for `ITask<T>`

Reply:
So the way to do that here is to do `return new FuncTask<ValidateGitInstallResult>(TaskEx.FromResult(new ValidateGitInstallResult(false, null, null)));`

This will create a completed task for FuncTask, and when that gets added to the task continuation, it'll automatically be done and won't start up a thread at all.